python_modules/dagster/dagster/_core/definitions/schedule_definition.py

```diff
@@ -584,16 +601,36 @@ def __init__(
                 " https://github.com/dagster-io/dagster/issues/15294 for more information."
             )
 
-        if job:
-            job_def = normalize_automation_target_def(job)
-            self._target = AutomationTarget(job_def)
+        if (
+            sum(
+                [
+                    int(target is not None),
+                    int(job_name is not None),
+                    int(job is not None),
+                ]
+            )
+            > 1
+        ):
+            raise DagsterInvalidDefinitionError(
+                "Attempted to provide more than one of 'job', 'job_name', and 'target'"
+                "params to ScheduleDefinition. Must provide only one."
+            )
+
+        if target:
+            self._target = AutomationTarget.from_coercible(
+                target,
+                automation_name=check.not_none(
+                    name, "If you specify target you must specify schedule name"
+                ),
+            )
+        elif job:
+            self._target = AutomationTarget.from_coercible(job)
         elif job_name:
             self._target = AutomationTarget(
                 resolvable_to_job=check.str_param(job_name, "job_name"),
-                op_selection=None,
             )
         else:
-            check.failed("Must provide job or job_name")
+            check.failed("Must provide target, job, or job_name")
```

Comment:
Extracting this into helper that returns a target could be cleaner. Not a necessary change so take it or leave it.

Reply:
Because the helper would need to take 4 params I prefer to keep it inline.

Reply:
sgtm